python/lsst/daf/butler/_registry_shim.py

```diff
@@ -281,7 +281,7 @@ def queryDatasetTypes(
         self,
         expression: Any = ...,
         *,
-        components: bool | None = None,
+        components: bool = False,
```

Comment:
When I was working on #937, it occurred to me that we really want to be warning if people set the components flag to False. What do you think about having this be something like:

```python
components : bool | _Sentinel = _DefaultSentinel
```
and then warning `if components is False` ?

Reply:
Yes, I was trying to say that we should warn any time it's not `_DefaultSentinel` because that would imply someone actually tried to set it. We should likely raise if they set it to anything other than False.